modelkit/assets/drivers/abc.py

```diff
@@ -3,13 +3,18 @@
 
 import pydantic
 
+from modelkit.core.settings import ModelkitSettings
 
-class StorageDriverSettings(pydantic.BaseSettings):
-    bucket: str = pydantic.Field(..., env="MODELKIT_STORAGE_BUCKET")
-    lazy_driver: bool = pydantic.Field(False, env="MODELKIT_LAZY_DRIVER")
 
-    class Config:
-        extra = "allow"
+class StorageDriverSettings(ModelkitSettings):
+    bucket: str = pydantic.Field(
+        ..., validation_alias=pydantic.AliasChoices("bucket", "MODELKIT_STORAGE_BUCKET")
```

Comment:
`ModelkitSettings` as well as `validation_alias=pydantic.AliasChoices(var_name, ENV_VAR_NAME)` are mandatory to keep the previous `env` behavior (now silently dropped from pydantic 2):
- if neither `var_name` nor the `ENV_VAR_NAME` are set, the default is used
- if the `var_name` is set, it is used: no matter whether `ENV_VAR_NAME` is set or not
- `ENV_VAR_NAME` is used if set and only if `var_name` is not set

I added a test showing this behavior: tests/test_settings.py

Reply:
Nicely done, tests are very easy to read.